Rework top section

jazeee opened the pull request:
Work in progress
===
@johnlamping - I am thinking we should reduce the clutter in the top area, and simplify the overhead around the editable vs read-only stuff.
See video for the direction I think we can go.

Summary
===
The reaction penalty is redundant with armor bonus.

The armor bonus can be negative or positive, and I often use it to manage boosts from buffs, or for negatives from stone skin or from temporary fumbles.

Removing that, opens up space for rework in the top bar.

EPN, CP, and FP don't need to clutter up the global area. Instead, the editable version should live on `Basic` and the read-only values can show up on spells and skills

Changes
===
- Remove reaction penalty
- Move EPN, CP, FP, Eff to tabs
- Move IQ, DX, BR to be a stack
- Adjust dice buttons.

Video
---

Initial iteration:

https://github.com/user-attachments/assets/49ef2451-1acc-466b-9ac6-951bc001c3b0

jazeee committed:
Rework top section

Summary
===
The reaction penalty is redundant with armor bonus.

The armor bonus can be negative or positive, and I often use it to manage boosts from buffs, or for negatives from stone skin or from temporary fumbles.

Removing that, opens up space for rework in the top bar.

EPN, CP, and FP don't need to clutter up the global area. Instead, the editable version should live on `Basic` and the read-only values can show up on spells and skills

Changes
===
- Remove reaction penalty
- Move EPN, CP, FP, Eff to tabs
- Move IQ, DX, BR to be a stack
- Adjust dice buttons.

jazeee committed:
Extract out power stats editable table and put it in basic tab

jazeee committed:
Add text to skills and spells

jazeee committed:
Remove EP, CP, from header

johnlamping commented:
This direction looks great!
I have a few suggestions, for a little more consistency:
When you are in the basic tab, make the IQ, DX, and BR on the top the same as on the other tabs. Instead, make the place where you edit them down below, where you edit EPN and CP. This gives up putting the edits next to the value, but it makes the top be consistent, lowering the cognitive load. Do the same thing with PWR, putting the editing down below in the basic tab.
Put the avail FP and CP in the same place on the basic tab as on the other tabs. (You could also repeat is where you edit them, but this way, people can also see them in the usual place.)
When the weight penalty moves to the equipment tab, maybe there should be some text next to it saying that it affects speed and defenses.

jazeee commented:
Here's some of the update, (not pushed yet)
![image](https://github.com/user-attachments/assets/a430be34-584b-4a24-b94b-0b0bc23f92f7)

jazeee committed:
Move editable IQ, DX, BR to basic tab

jazeee committed:
Rework the top section buttons

jazeee committed:
Move general roll buttons to Skills

jazeee committed:
Adjust columns in overview

jazeee committed:
Consolidate editing into basic tab

jazeee committed:
Move IQ, DX, BR to overview tab

jazeee committed:
Rearrange rows

jazeee committed:
Remove redundant code

jazeee committed:
Remove unused buttons (do/undo top actions)

jazeee committed:
Fix no-add-modify-css

jazeee committed:
Add utility to update roll section

jazeee committed:
Add overview skill to roll button

johnlamping commented:
You've been busy!

jazeee committed:
Add feat builder

jazeee committed:
Convert to advance boost

jazeee commented:
@johnlamping - making progress:

https://github.com/user-attachments/assets/f7da48d6-bac3-42b8-8b6f-aa177795fa5d

jazeee committed:
Move roll button to Notes

jazeee committed:
Wire up IQ, DX, BR, roll to roll section

jazeee committed:
Fix case

jazeee committed:
Add CP and FP available to table headers

jazeee committed:
Convert spell header buttons

jazeee committed:
Add button to arcane spells

jazeee committed:
Add spell pwr to arcane updater

jazeee committed:
Rearrange weight content

jazeee commented:
@johnlamping 
More updates. I think for this iteration, since it is getting large, we may want to scope the rest of the work to:
1. Add rest of defense table
2. Identify where initiative and speed should go.

https://github.com/user-attachments/assets/9acca7a0-c749-4fc5-921c-cbbc5d8b3973

johnlamping commented:
This looks great!

The only issue I noticed is that the ? that you can click to see the weight thresholds feels like it should be in the equipment tab, next to the weight penalty, not up by speed like it is now.

Also, would it make sense to put the simple roll, IQ roll, etc, in their own column? That way they would save some vertical space, and it would make them clearly different from skills.

jazeee commented:
Agreed on all points.

On Sun, Mar 9, 2025, 7:55 PM John Lamping ***@***.***> wrote:

> This looks great!
>
> The only issue I noticed is that the ? that you can click to see the
> weight thresholds feels like it should be in the equipment tab, next to the
> weight penalty, not up by speed like it is now.
>
> Also, would it make sense to put the simple roll, IQ roll, etc, in their
> own column? That way they would save some vertical space, and it would make
> them clearly different from skills.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2709304392>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK6O2G73YGALCA7VUZUCTD2TUELVAVCNFSM6AAAAABXW66I2OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMBZGMYDIMZZGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: johnlamping]*johnlamping* left a comment
> (epic-power-rpg/roll20#121)
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2709304392>
>
> This looks great!
>
> The only issue I noticed is that the ? that you can click to see the
> weight thresholds feels like it should be in the equipment tab, next to the
> weight penalty, not up by speed like it is now.
>
> Also, would it make sense to put the simple roll, IQ roll, etc, in their
> own column? That way they would save some vertical space, and it would make
> them clearly different from skills.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2709304392>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK6O2G73YGALCA7VUZUCTD2TUELVAVCNFSM6AAAAABXW66I2OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMBZGMYDIMZZGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

johnlamping commented:
One more thing, the arrow that converts ER to PWR is even more confusing than before, since it now looks like all those other upward arrows that do something entirely different. I don't have a suggestion for fixing it that I really like. It could get replaced by one of these: ☝,  ╨, or ⇈. Maybe this character, to the left of the ER text, would convey the idea, and wouldn't be so confusing: ╙

jazeee commented:
Yeah, I couldn't think of a good character.

On Sun, Mar 9, 2025, 8:14 PM John Lamping ***@***.***> wrote:

> One more thing, the arrow that converts ER to PWR is even more confusing
> than before, since it now looks like all those other upward arrows that do
> something entirely different. I don't have a suggestion for fixing it that
> I really like. It could get replaced by one of these: ☝, ╨, or ⇈. Maybe
> this character, to the left of the ER text, would convey the idea, and
> wouldn't be so confusing: ╙
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2709323811>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK6O2ASFU4JJW74H2WLQPD2TUGTHAVCNFSM6AAAAABXW66I2OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMBZGMZDGOBRGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: johnlamping]*johnlamping* left a comment
> (epic-power-rpg/roll20#121)
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2709323811>
>
> One more thing, the arrow that converts ER to PWR is even more confusing
> than before, since it now looks like all those other upward arrows that do
> something entirely different. I don't have a suggestion for fixing it that
> I really like. It could get replaced by one of these: ☝, ╨, or ⇈. Maybe
> this character, to the left of the ER text, would convey the idea, and
> wouldn't be so confusing: ╙
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2709323811>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK6O2ASFU4JJW74H2WLQPD2TUGTHAVCNFSM6AAAAABXW66I2OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMBZGMZDGOBRGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

jazeee committed:
Convert to finger up icon

jazeee commented:
Finger up icon
![image](https://github.com/user-attachments/assets/f3598095-d397-481f-91c0-c44cec3887ea)

jazeee committed:
Add speed to overview tab

jazeee committed:
Move speed tooltip to weight section

jazeee committed:
Clean up weight penalties popup

jazeee commented:
![image](https://github.com/user-attachments/assets/ad0acd14-d7e8-4d4d-bebd-00ce93355033)

jazeee committed:
Move Attribute rolls to left column

jazeee commented:
![image](https://github.com/user-attachments/assets/842e2ce5-ed00-469e-b56c-9401a8243035)

jazeee committed:
Move feats up

jazeee commented:
![image](https://github.com/user-attachments/assets/9fd965ef-9af5-4bd3-8bae-38e8f184e9e3)

jazeee commented:
I think the Overview tab is where people will go to set up their rolls. (outside of spells)

To that end, `Initiative` should be there too.
I also put speed in the Advantages section.

jazeee commented:
![image](https://github.com/user-attachments/assets/cdcb7f9c-4146-4f93-8450-59277896aa60)

jazeee committed:
Move initiative to overview

jazeee committed:
Add speed to equipment

jazeee commented:
![image](https://github.com/user-attachments/assets/34b794bb-7867-479c-9d57-751f1ca53a16)

jazeee commented:
@johnlamping I think one more weekend of updates will get us to a working state.
Feel free to try it out. Thoughts:
1. For non-casters, the Overview tab is your action builder.
   1. You assemble a combination of skills, and feats to build your action.
   2. Then you decide how much to boost, then roll
1. For casters, the spell tab is where you build your spell casting.
   1. This needs more adjustments.
   2. The action builder concept may need to also show up here. "Cast with COW", and/or another feat
   3. For arcane, it may be set
   4. For divine, I think we need to add a `Deity` field, and when you click on the action button, the description becomes "Pray to deity - Divine Healing"
   5. For innate, the action button could just update the description.
1. The power and defense section is now clean, ready for guard, robustness, resolve.
   1. For the other defenses, we calculate these fields based on armor.
   2. These 3 fields are different.... Need to think about the implementation.
1. We should hide some fields in the Overview tab (Defense skills)

It's getting there.

johnlamping commented:
It's looking great. I'll try to run it soon.

My only comment right now is that the upward pointing hand, ☝, isn't working for me. (I know ... I suggested it.) For one thing, it's really hard to recognize in Roll20's font. Maybe the version with the text variant, ☝︎, would work better, or the upward pointing back of the hand, 👆.
I'd love it if there were some character that suggested copying, but the closest I could find was a clipboard, 📋, which doesn't convey much of anything. This might be good if the arrow pointed upwards ⎘.

jazeee commented:
We may want to use color as well.
A blue variant plus a blue hue in the header may link the sections.
A running man icon may help indicate action.

On Sun, Mar 16, 2025, 8:15 PM John Lamping ***@***.***> wrote:

> It's looking great. I'll try to run it soon.
>
> My only comment right now is that the upward pointing hand, ☝, isn't
> working for me. (I know ... I suggested it.) For one thing, it's really
> hard to recognize in Roll20's font. Maybe the version with the text
> variant, ☝︎, would work better, or the upward pointing back of the hand, 👆.
> I'd love it if there were some character that suggested copying, but the
> closest I could find was a clipboard, 📋, which doesn't convey much of
> anything. This might be good if the arrow pointed upwards ⎘.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2727931125>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK6O2FQWHBLCXQ2TDTMTAT2UY435AVCNFSM6AAAAABXW66I2OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMRXHEZTCMJSGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: johnlamping]*johnlamping* left a comment
> (epic-power-rpg/roll20#121)
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2727931125>
>
> It's looking great. I'll try to run it soon.
>
> My only comment right now is that the upward pointing hand, ☝, isn't
> working for me. (I know ... I suggested it.) For one thing, it's really
> hard to recognize in Roll20's font. Maybe the version with the text
> variant, ☝︎, would work better, or the upward pointing back of the hand, 👆.
> I'd love it if there were some character that suggested copying, but the
> closest I could find was a clipboard, 📋, which doesn't convey much of
> anything. This might be good if the arrow pointed upwards ⎘.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2727931125>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK6O2FQWHBLCXQ2TDTMTAT2UY435AVCNFSM6AAAAABXW66I2OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMRXHEZTCMJSGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

johnlamping commented:
Now you are showing me up with your UI expertise. Sounds right, though!

jazeee committed:
Update to new color

jazeee committed:
Add action button

jazeee committed:
Adjust action area to blue hues

jazeee commented:
@johnlamping this adjusts the action update buttons
![image](https://github.com/user-attachments/assets/b1145757-d77c-4848-be41-9b27c0e96319)
![image](https://github.com/user-attachments/assets/9987b389-331f-4838-961b-e5558fb61f69)

jazeee committed:
Clean up pwr-er-hp section

jazeee commented:
And cleaning up PWR section
![image](https://github.com/user-attachments/assets/17531369-803c-46cd-9ca3-53986055237c)

johnlamping commented:
That's looking fantastic!

jazeee committed:
Add resolve, robust defenses

jazeee committed:
Add toggle control for defenses

jazeee commented:
@johnlamping here is the defense table solution I added (toggle between tables)
I also added some thoughts/questions to my video.
We're closing in on a place where we should deploy it.
Some thoughts:
1. Divine needs a button to update skill section for each spell
2. Divine needs a field for the deity.
3. The area above the tabs could be used for temporary boosts/drains, such as: `IQ boost` `DX Boost`, `BR boost`, `To hit boost` (and the like)
4. Innate spells aren't boostable, so perhaps they just need a "cast" button.

https://github.com/user-attachments/assets/47cdf0f6-5908-4c40-8865-67eff57bf189

johnlamping commented:
Closing in on deployment!

For a boost section on the left, how about having three rows, each with a drop-down to select what the boost/penalty applies to (IQ, DX, Dodge, Resolve, All defenses, etc) and a box with the value of that boost/penalty? That way, a player could keep track of whatever boosts came up, without making the area cluttered. You probably also want a "---" value, that says you are not boosting anything with that row. In the HTML, you could have an invisible box holding the current value of each kind of boost. Whenever some piece of code cares about a particular boost, it looks at the appropriate invisible box. Then you have a script that run whenever any boosts change. It first sets the invisible boxes' values to 0, and then iterates through the boost rows, setting the hidden boxes for the boosts that are affected.

An easy way to handle Divine would be to put a row above the spells that echos the prayer skill, and has the button to copy it to the top. Somewhat more work would be to have the copy button for each spell copy the spell name to the top row of the roll area, and copy the prayer ability to the bottom row of that area.

For innate spells, the button by the spell might put the spell on the top row of the roll area, and put Spell Hit in the bottom row. (I've been meaning to change the Touch/Ray/Missile column to Touch/Ray/Missile/Area/Mental/Affliction. But I didn't want to do that in the middle of all your changes. But with that change, the innate spell would copy whatever attack is used for that spell.)

johnlamping commented:
Maybe, instead of
⇑ ER to PWR ⇑
it might be better as
1 ER ↝ 2 PWR
That way, the double arrow always means do something with the whole number, and this suggests it's something different.

jazeee committed:
Update button text

jazeee commented:
`1 ER ↝ 2 PWR` sounds great. Updated

jazeee commented:
> An easy way to handle Divine would be to put a row above the spells that echos the prayer skill, and has the button to copy it to the top. Somewhat more work would be to have the copy button for each spell copy the spell name to the top row of the roll area, and copy the prayer ability to the bottom row of that area.

The button on each row should be easy enough.

jazeee committed:
Add deity row (Divine can have multiple deities)

jazeee commented:
Added divine deity:
![image](https://github.com/user-attachments/assets/06d5f7f1-c3ec-4886-9860-2d1286817c5a)

jazeee committed:
Clean up css

johnlamping commented:
I like the deity row.

jazeee committed:
Add pray update action area

jazeee commented:
Divine casting is in:

https://github.com/user-attachments/assets/ed80fd36-f3f4-4b59-9aa4-157734d84954

jazeee commented:
@johnlamping the innate spells section are not yet updated, and the top right panel is still open.
I recommend we get this QA'd and possibly merged. It is a large set of changes.
We can do follow up on the remaining work, either as a stacked branch, or a follow up. (I may look at it later in the evening)

johnlamping commented:
I spent maybe an hour testing things. Here are all the issues I noticed. I'll mark the change as approved, so you can merge, but I think it is best to fix the first two issues below before trying this in the campaign.

I would consider these blocking, because they are significant inconsistencies:
- In the overview tab, when feats are copied, the PWR cost in the rolls section isn't getting zeroed out, like it is for skills.
- For arcane spells, the if the PWR cost in the spell listing isn't a valid number, the PWR cost in the rolls section isn't being changed. It should be zeroed out, which is how it works for the divine section.

These are ones that I think are worth changing, but I could be argued out of. And I don't think they're blocking.
- When a copy button is clicked, the mods are getting copied to the boost in the roll section. I, at least, tend to use them to record long standing buffs, like from a high quality weapon, or the like. And, now with rolling happening in the top section, there is a place to put a boost, if desired. I think it might be better to copy the mods to the first mod field in the roll section.
- The ? button that brings up the weight penalty table should be next to the weight penalty, not to the right of speed, like it is now.
- When the weight penalty is 0, it shows as blank, which I found confusing. I think putting in a 0 would be clearer.

These are some nice-to-haves.
- Maybe put a rounded border around the top roll section to echo the icons that fill it in.
- In the Overview tab, maybe put the list of advantages and disadvantages to the right hand column. It will be more visible there, and everything else in the other two columns has rolls.
- In the Divine tab, the prayer ability isn't getting copied up to the ability in the rolls section.

johnlamping commented:
A thought about how to eventually handle innate spells: Make it so the ability number in the roll section can be set to "--", which means not defined, the same as it does in the skills section if you haven't set the base ability to IQ, DX, etc. When it is set that way, you still can push the roll button. It would charge PWR and EP according to the top row, and it would chat a message, but it wouldn't roll or charge PWR for the boost. 
When you copy an innate spell, it puts the spell on the top row, and sets the ability to "--", activating this feature.
You might even make it so that the character in the roll button changes from two squqres to something else when the ability is "--"

jazeee commented:
Very helpful. Thanks.
For feats, I was intentionally thinking that they build in the description,
on top of skills.
An action builder.

1. Pick your skill. Let's say, attack with axe
2. Add feats, like clash or heavy hit. As you add them, the description
accumulates, automatically adding per costs.

In doing so, people can prepare their next attack and see the cost

Similarly, for a spell, one could do the same, although you'd have to pick
the spell first, then, pick feats.

Once built, you could save to favorites.

It would help new players, and also help GMs play multiple monsters.

I'll take a look at the items. The qa is really helpful.

On Sun, Mar 23, 2025, 4:06 PM John Lamping ***@***.***> wrote:

> A thought about how to eventually handle innate spells: Make it so the
> ability number in the roll section can be set to "--", which means not
> defined, the same as it does in the skills section if you haven't set the
> base ability to IQ, DX, etc. When it is set that way, you still can push
> the roll button. It would charge PWR and EP according to the top row, and
> it would chat a message, but it wouldn't roll or charge PWR for the boost.
> When you copy an innate spell, it puts the spell on the top row, and sets
> the ability to "--", activating this feature.
> You might even make it so that the character in the roll button changes
> from two squqres to something else when the ability is "--"
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2746531572>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK6O2GQIRO6CL4UPZYU4E32V45AFAVCNFSM6AAAAABXW66I2OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONBWGUZTCNJXGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: johnlamping]*johnlamping* left a comment
> (epic-power-rpg/roll20#121)
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2746531572>
>
> A thought about how to eventually handle innate spells: Make it so the
> ability number in the roll section can be set to "--", which means not
> defined, the same as it does in the skills section if you haven't set the
> base ability to IQ, DX, etc. When it is set that way, you still can push
> the roll button. It would charge PWR and EP according to the top row, and
> it would chat a message, but it wouldn't roll or charge PWR for the boost.
> When you copy an innate spell, it puts the spell on the top row, and sets
> the ability to "--", activating this feature.
> You might even make it so that the character in the roll button changes
> from two squqres to something else when the ability is "--"
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2746531572>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK6O2GQIRO6CL4UPZYU4E32V45AFAVCNFSM6AAAAABXW66I2OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONBWGUZTCNJXGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

johnlamping commented:
Oh, that makes lots of sense about how feats work!

To be consistent, the PWR and ER field should then go down on the bottom row with skills, since they get set when you set when you set that row. But there is no room for them there. 

Maybe this reorganization might work (not urgent!):
Move Skill to the top row, because the feats are subsidiary to it. 
The top row becomes Skill, ability, PWR, ER, boost: all the things that you don't clear when you add a feat.
The bottom row becomes Description (or Feats) Feat PWR, a bit of a gap, mods (which should line up under boost).
To the right of both rows, centered vertically, goes calculated PWR.

jazeee commented:
Yeah, makes sense. I could also use colors to separate skills from
description/feats. Blue button for feats, and green or yellow for
description.

On Sun, Mar 23, 2025, 4:38 PM John Lamping ***@***.***> wrote:

> Oh, that makes lots of sense about how feats work!
>
> To be consistent, the PWR and ER field should then go down on the bottom
> row with skills, since they get set when you set when you set that row. But
> there is no room for them there.
>
> Maybe this reorganization might work (not urgent!):
> Move Skill to the top row, because the feats are subsidiary to it.
> The top row becomes Skill, ability, PWR, ER, boost: all the things that
> you don't clear when you add a feat.
> The bottom row becomes Description (or Feats) Feat PWR, a bit of a gap,
> mods (which should line up under boost).
> To the right of both rows, centered vertically, goes calculated PWR.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2746552608>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK6O2AF2CKMG7UF6N3HAX32V5AW3AVCNFSM6AAAAABXW66I2OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONBWGU2TENRQHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: johnlamping]*johnlamping* left a comment
> (epic-power-rpg/roll20#121)
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2746552608>
>
> Oh, that makes lots of sense about how feats work!
>
> To be consistent, the PWR and ER field should then go down on the bottom
> row with skills, since they get set when you set when you set that row. But
> there is no room for them there.
>
> Maybe this reorganization might work (not urgent!):
> Move Skill to the top row, because the feats are subsidiary to it.
> The top row becomes Skill, ability, PWR, ER, boost: all the things that
> you don't clear when you add a feat.
> The bottom row becomes Description (or Feats) Feat PWR, a bit of a gap,
> mods (which should line up under boost).
> To the right of both rows, centered vertically, goes calculated PWR.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2746552608>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK6O2AF2CKMG7UF6N3HAX32V5AW3AVCNFSM6AAAAABXW66I2OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONBWGU2TENRQHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

johnlamping commented:
As long as you are the one doing the work, the copy buttons for feats could look slightly different from the other ones, too. Maybe they could have a + next to their arrow, to indicate they are adding, rather than replacing, like the other ones.

jazeee commented:
AI is helping with the SVG icons, lol.
Good idea on the plus button.

On Sun, Mar 23, 2025, 5:26 PM John Lamping ***@***.***> wrote:

> As long as you are the one doing the work, the copy buttons for feats
> could look slightly different from the other ones, too. Maybe they could
> have a + next to their arrow, to indicate they are adding, rather than
> replacing, like the other ones.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2746583301>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK6O2FCB2TK2FY6PBCC3FD2V5GMRAVCNFSM6AAAAABXW66I2OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONBWGU4DGMZQGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: johnlamping]*johnlamping* left a comment
> (epic-power-rpg/roll20#121)
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2746583301>
>
> As long as you are the one doing the work, the copy buttons for feats
> could look slightly different from the other ones, too. Maybe they could
> have a + next to their arrow, to indicate they are adding, rather than
> replacing, like the other ones.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/epic-power-rpg/roll20/pull/121#issuecomment-2746583301>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK6O2FCB2TK2FY6PBCC3FD2V5GMRAVCNFSM6AAAAABXW66I2OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONBWGU4DGMZQGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

jazeee commented:
> * For arcane spells, the if the PWR cost in the spell listing isn't a valid number, the PWR cost in the rolls section isn't being changed. It should be zeroed out, which is how it works for the divine section.

and

> When a copy button is clicked, the mods are getting copied to the boost in the roll section. I, at least, tend to use them to record long standing buffs, like from a high quality weapon, or the like. And, now with rolling happening in the top section, there is a place to put a boost, if desired. I think it might be better to copy the mods to the first mod field in the roll section.

and

>The ? button that brings up the weight penalty table should be next to the weight penalty, not to the right of speed, like it is now.
> When the weight penalty is 0, it shows as blank, which I found confusing. I think putting in a 0 would be clearer.

Fixed here, along with mod2,mod3 - should we do mod2-mod3 or all 3?
https://github.com/epic-power-rpg/roll20/pull/124

This is a small stacked change on this branch

jazeee commented:
@johnlamping For
> In the Divine tab, the prayer ability isn't getting copied up to the ability in the rolls section.

This requires the deity name to be in the skill section like this:
![image](https://github.com/user-attachments/assets/254e27a5-8b71-4a14-8043-ff19a3221566)

I guess, we may want to adjust this to exact match, since `Thor's hammer` would conflict with the Deity.

Unsure really. We can easily adjust this, of course.

jazeee committed:
Fix number parsing for invalid chars (#124)

* Fix number parsing for invalid chars

* Restore save order of roll keys and don't update roll_mod2, roll_mod3

* Fix weight penalty and tooltip

* Add gap
